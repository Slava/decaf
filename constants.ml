open Core.Std

type decaf_constant =
  | IntConstant of int
  | DoubleConstant of float
  | StringConstant of string
  | BoolConstant of bool
  | CharConstant of char
;;

let stringify_constant c =
  match c with
  | IntConstant i -> string_of_int i
  | DoubleConstant f -> Float.to_string f
  | StringConstant s -> "\"" ^ s ^ "\""
  | BoolConstant b -> string_of_bool b
  | CharConstant c -> String.of_char c
;;
