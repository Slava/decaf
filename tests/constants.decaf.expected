Program
  Function int: Constants:
    (1)
    ("something something")
    ('a')
    ('1')
    (true)
    (false)
    (null)
